(GH-2467) Clarify using interpolations in plan_hierarchy (#2576)
puppetlabs/bolt: documentation/writing_plans.md

}
```

### Using interpolations in the `plan_hierarchy` hierarchy

> Interpolations are not well supported in the `plan_hierarchy` hierarchy. Puppet plans run in a
> custom, temporary Puppet environment and are not run in the context of a particular target.

Reviewer: ```suggestion
Interpolations are not well supported in the `plan_hierarchy` hierarchy. Bolt runs 
Puppet plans in a custom, temporary Puppet environment, and does not run the plans 
in the context of a particular target.
```

> Therefore target level data such as facts will not be available to look ups outside of apply blocks,

Reviewer: ```suggestion
Target level data such as facts are not available to lookups outside of apply blocks,
```
I think "lookup" is noun here, so one word is fine

> so the normal hierachy interpolation will not work. The `lookup()` function will interpolate based on
> variables currently in scope.

Reviewer: ```suggestion
so the normal hierarchy interpolation does not work. The `lookup()` function 
only interpolates based on the variables currently in scope.
```

The following example demonstrates interpolating the `$application` variable into a `plan_hierarchy`
hierarchy:

Using this `hiera.yaml`:
```yaml
version: 5

plan_hierarchy:
  - name: Application Data
    path: %{application}.yaml
```

With this data at `<PROJECT DIRECTORY>/data/kittycats.yaml`:
```
site_path: /var/www/kittycats.tld/public_html
```

And this data at `<PROJECT DIRECTORY/data/doggos.yaml`:
```
site_path: /var/www/doggos.tld/public_html
```

Bolt looks up the site path and passes the value to a task to deploy the site.
```
plan plan_lookup(
  TargetSpec $targets,
  String $application = 'doggos'
) {
  $site_path = lookup('site_path')
  run_task("deploy_site", $targets, 'path' => $site_path)
}
```

> Note that if you try to call `lookup('site_path')` from a subplan of `plan_lookup`, like so:

Reviewer: ```suggestion
Note that if you tried to call `lookup('site_path')` from a subplan of `plan_lookup`, like so:
```

```
plan plan_lookup(
  TargetSpec $targets,
  String $application = 'doggos'
) {
  $site_path = lookup('site_path')
  run_task("deploy_site", $targets, 'path' => $site_path)
}
```

```
plan other_plan(
  TargetSpec $targets
) {
  lookup('site_path')
  ...
}
```

> This will error, as `$application` is not in scope in `other_plan`.

Reviewer: ```suggestion
This would error, because `$application` is not in scope in `other_plan`.
```
